### PR TITLE
Add pylint

### PR DESCRIPTION
**Pull request by Paulinakhew**
- add [pylint](https://docs.pylint.org/en/1.6.0/intro.html), a tool for checking code 

### WHAT is this pull request doing?
- adding pylint as a pre-commit hook that is used in CI
- `pylint` is used by the [Google Python Style guide](https://google.github.io/styleguide/pyguide.html) which is used in the [Shopify Python guide](https://github.com/Shopify/shopify_python)
- the `pylintrc` file added is the one from the Google style guide with the exception of:
  - `max-line-length=120` (Shopify style guide asks for 120)
  - `confidence=INFERENCE` (ignore small changes that pylint isn't too confident about)
  - `ignore=missing-class-docstring,missing-module-docstring` (most modules/classes don't have docstrings)
- I manually added a couple of changes

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -14,3 +14,7 @@ repos:
     rev: 20.8b1
     hooks:
     -   id: black
```
Comment (paulomarg) [inline .pre-commit-config.yaml:16]: Do we still need black / autopep? Not a problem if we do, just checking!

**Review by paulomarg — APPROVED** [review]
This all makes sense to me!

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Paulinakhew) [reply .pre-commit-config.yaml:16]: I think we still do so future PRs have the same formatting that black/autopep suggest! Black has a very specific code style so enforcing that going forward makes the code more readable

Comment (paulomarg) [reply .pre-commit-config.yaml:16]: That makes sense.

**Pull request merged**